docs/typescript.md

```diff
@@ -0,0 +1,221 @@
+---
+id: typescript
+title: Using TypeScript with React Native
+---
+
+[TypeScript][ts] is a language which extends JavaScript to add type definitions, a lot like [Flow][flow] which React Native is built in. React Native supports TypeScript and Flow out of the box by default.
+
+## Getting Started with TypeScript
+
+If you're starting a new project, you can use the [TypeScript template][ts-template] to get you there faster:
+
+```sh
+react-native init MyAwesomeTSProject --template typescript
+```
+
+Alternatively, you can use some of the community bootstrapping tools which can get you to a complete app faster:
```

Comment:
small nit on wording - is `react-native-cli` not a community bootstrapping tool as well? https://github.com/react-native-community/cli

Reply:
Hah, I guess it is given that it lives in there